core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/internal/signer/V4RequestSigner.java

```diff
@@ -52,9 +55,10 @@ static V4RequestSigner header(V4Properties properties) {
                 requestBuilder.putHeader(SignerConstant.X_AMZ_SECURITY_TOKEN,
                                          ((AwsSessionCredentialsIdentity) properties.getCredentials()).sessionToken());
             }
+            addHostHeader(requestBuilder);
             addDateHeader(requestBuilder, formatDateTime(properties.getCredentialScope().getInstant()));
 
-            V4RequestSigningResult result = create(properties).sign(requestBuilder);
+            V4RequestSigningResult result = create(properties, getContentHash(requestBuilder)).sign(requestBuilder);
```

Comment:
thinking aloud: should the contentHash be a constructor input like properties, or input to `sign()`?

Reply:
It's easier if it's a constructor param - sign's signature would change, which is a bigger change. If we decide that `x-amz-content-sha256` shouldn't be added as a header unless required (s3), then we'll likely need to refactor it as an input to request-signer and payload-signer